test/unit/math/prim/arr/functor/coupled_ode_observer_test.cpp

```diff
@@ -0,0 +1,73 @@
+#include <stan/math/prim/arr.hpp>
+#include <gtest/gtest.h>
+#include <test/unit/util.hpp>
+#include <test/unit/math/prim/arr/functor/harmonic_oscillator.hpp>
+#include <test/unit/math/prim/arr/functor/mock_ode_functor.hpp>
+#include <test/unit/math/prim/arr/functor/mock_throwing_ode_functor.hpp>
+#include <vector>
+#include <string>
+
+struct StanMathOde : public ::testing::Test {
+  std::stringstream msgs;
+  std::vector<double> x;
+  std::vector<int> x_int;
+};
+
+TEST_F(StanMathOde, observe_states_dd) {
```

Comment:
Can we change the name of the test? There are now 4 input that can be passed as dbl or var. So here, a title such as `observe_states_dddd` may be more appropriate.

Reply:
Let me have a look...but i would like to avoid having to change all our test names....

Reply:
I suggest either using the `dddd` suffix, or something more specific like `all_double`. I get what you're doing, but some fresh eyes may not.

Reply:
sure... let's do this convention in the observer tests.